buildenv/docker/mkdocker.sh

```diff
@@ -29,42 +29,43 @@ usage() {
-  echo "  --build                build the docker image (overrides '--print')"
-  echo "  --cuda                 include CUDA header files"
-  echo "  --dist=...             specify the Linux distribution (e.g. centos, ubuntu)"
-  echo "  --freemarker           include freemarker.jar"
-  echo "  --gitcache={yes|no}    generate the git cache (default: yes)"
-  echo "  --jdk=...              specify which jdks can be built (default: all)"
-  echo "  --print                write the Dockerfile to stdout (default; overrides '--build')"
-  echo "  --tag=...              specify a name for the docker image (may be repeated, default: none)"
-  echo "  --user=...             specify the user name (default: 'jenkins')"
-  echo "  --version=...          specify the distribution version (e.g. 6, 16.04)"
+  echo "  --help|-h             print this help, then exit"
+  echo "  --arch=...            specify the processor architecture (default: host architecture)"
+  echo "  --build               build the docker image (overrides '--print')"
+  echo "  --cuda                include CUDA header files"
+  echo "  --dist=...            specify the Linux distribution (e.g. centos, ubuntu)"
+  echo "  --freemarker          include freemarker.jar"
+  echo "  --gitcache={yes|no}   generate the git cache (default: yes)"
+  echo "  --jdk=...             specify which JDKs can be built (default: all)"
+  echo "  --print               write the Dockerfile to stdout (default; overrides '--build')"
+  echo "  --tag=...             specify a name for the docker image (may be repeated, default: none)"
+  echo "  --user=...            specify the user name (default: 'jenkins')"
+  echo "  --version=...         specify the distribution version (e.g. 6, 18.04)"
   echo ""
   local arch="$(uname -m)"
   echo "Supported build patterns on this host ($arch):"
 if [ $arch = x86_64 ] ; then
-  echo "  bash mkdocker.sh --tag=openj9/cent610 --dist=centos --version=6 --build"
+  echo "  bash mkdocker.sh --tag=openj9/cent6 --dist=centos --version=6  --build"
 fi
 if [ $arch = x86_64 -o $arch = ppc64le ] ; then
-  echo "  bash mkdocker.sh --tag=openj9/cent7  --dist=centos --version=7   --build"
+  echo "  bash mkdocker.sh --tag=openj9/cent7 --dist=centos --version=7  --build"
 fi
-  echo "  bash mkdocker.sh --tag=openj9/ub16   --dist=ubuntu --version=16  --build"
-  echo "  bash mkdocker.sh --tag=openj9/ub18   --dist=ubuntu --version=18  --build"
-  echo "  bash mkdocker.sh --tag=openj9/ub20   --dist=ubuntu --version=20  --build"
+  echo "  bash mkdocker.sh --tag=openj9/ub16  --dist=ubuntu --version=16 --build"
+  echo "  bash mkdocker.sh --tag=openj9/ub18  --dist=ubuntu --version=18 --build"
+  echo "  bash mkdocker.sh --tag=openj9/ub20  --dist=ubuntu --version=20 --build"
   exit 1
 }
 
 # Global configuration variables.
 action=print
+all_versions=
 arch=
-bootjdk_versions=
 cuda=no
 dist=unspecified
 freemarker=no
 gen_git_cache=yes
 jdk_versions=all
+declare -A known_version
```

Comment:
Probably don't need this to be global considering it's only used in `validate_options`

Reply:
Fixed in f5889f7e1c337a7ff88cd70d2a310e43082790a3.